test/support/integration/teams_server.ex

```diff
@@ -0,0 +1,299 @@
+  end
+
+  defp ensure_teams_key(state) do
+    if state.teams_key,
+      do: state,
+      else: %{
+        state
+        | teams_key: call_erpc_function(state.node, :create_org_key, [[org: state.org]]).key_hash
+      }
+  end
+
+  defp start_app(state) do
+    env =
+      for {key, value} <- env(state), into: [] do
+        {String.to_charlist(key), String.to_charlist(value)}
+      end
+
+    args = [
+      "-e",
+      "spawn(fn -> IO.gets([]) && System.halt(0) end)",
+      "--sname",
+      to_string(state.node),
+      "--cookie",
+      to_string(Node.get_cookie()),
+      "-S",
+      "mix",
+      "phx.server"
+    ]
+
+    port =
+      Port.open({:spawn_executable, elixir_executable()}, [
+        :exit_status,
+        :use_stdio,
+        :stderr_to_stdout,
+        :binary,
+        :hide,
+        env: env,
+        cd: app_dir(),
+        args: args
+      ])
+
+    wait_on_start(state, port)
+  end
+
+  defp fetch_url(state) do
+    port = state.app_port || app_port()
+    "http://localhost:#{port}"
+  end
+
+  defp prepare_database(state) do
+    :ok = mix(state, ["ecto.drop", "--quiet"])
+    :ok = mix(state, ["ecto.create", "--quiet"])
+    :ok = mix(state, ["ecto.migrate", "--quiet"])
+  end
+
+  defp ensure_app_dir! do
+    dir = app_dir()
+
+    unless File.exists?(dir) do
+      IO.puts(
+        "Unable to find #{dir}, make sure to clone the hub repository " <>
+          "into it to run integration tests or set TEAMS_PATH to its location"
+      )
+
+      System.halt(1)
+    end
+  end
+
+  defp app_dir do
+    System.get_env("TEAMS_PATH", @default_teams_dir)
+  end
+
+  defp app_port do
+    System.get_env("TEAMS_PORT", "4043")
+  end
+
+  defp debug do
+    System.get_env("TEAMS_DEBUG", "false")
+  end
+
+  defp proto do
+    System.get_env("TEAMS_LIVEBOOK_PROTO_PATH")
+  end
+
+  defp wait_on_start(state, port) do
+    url = state.url || fetch_url(state)
+
+    case :httpc.request(:get, {~c"#{url}/public/health", []}, [], []) do
+      {:ok, _} ->
+        port
+
+      {:error, _} ->
+        Process.sleep(10)
+        wait_on_start(state, port)
+    end
+  end
+
+  defp mix(state, args) when is_struct(state) do
+    state |> env() |> mix(args)
+  end
+
+  defp mix(env, args) do
+    cmd_opts = [
+      stderr_to_stdout: true,
+      env: env,
+      cd: app_dir(),
+      into: IO.stream(:stdio, :line)
+    ]
+
+    args = ["--erl", "-elixir ansi_enabled true", "-S", "mix" | args]
+
+    case System.cmd(elixir_executable(), args, cmd_opts) do
+      {_, 0} -> :ok
+      _ -> :error
+    end
+  end
+
+  defp env(state) do
+    app_port = state.app_port || app_port()
+    env(app_port, state.env)
+  end
+
+  defp env(app_port, state_env) do
+    env = %{
+      "MIX_ENV" => "livebook",
+      "PORT" => to_string(app_port),
+      "DEBUG" => debug()
+    }
+
+    env = if proto(), do: Map.merge(env, %{"LIVEBOOK_PROTO_PATH" => proto()}), else: env
```

Comment:
do we need to configure proto path? I think it'd always be this?

```suggestion
      "DEBUG" => debug(),
      "LIVEBOOK_PROTO_PATH" => "./proto"
    }
```

edit: actually do we even use proto in LT?

Reply:
Not yet, I can remove for now.